microsoft/PowerToys
# [FZ Editor] Serialize/deserialize settings

## Summary of the Pull Request

Get rid of temp files, serialize/deserialize `zones-settings.json` in the editor.

## PR Checklist
* [x] Applies to #8614

## Info on Pull Request

Removed temp files usage. Editor reads and writes data directly to `zones-settings.json`.
For parsing there is a 10ms retry loop with a 1-second timeout and a message box error if after 1 second the load/save fails.
If an error occurred:

- `Yes` will open the editor ignoring malformed data and using default values where necessary. `zones-settings.json` would be changed and contain only correct data.
- `No` closes the editor, data in `zones-settings.json` wouldn't be changed.

## Validation Steps Performed

- Apply layout and verify that it was applied by FZ correctly.
- Run unit tests.
- Change `zones-settings.json` manually and verify that incorrect data leads to an error message. 
- Verify behavior with `Yes` and `No` scenarios.
